### PR TITLE
mesh: Fix memory leaks

### PR DESCRIPTION
**Pull request by forgithub123**
Function return without os_mbuf_free_chain(buf) after buf = NET_BUF_SIMPLE(n)
will cause memory leaks

### CODE REVIEW

File: nimble/host/mesh/src/cfg_srv.c [hunk 1]
```diff
@@ -735,6 +735,7 @@ static void beacon_set(struct bt_mesh_model *model,
 		BT_ERR("Unable to send Config Beacon Status response");
 	}
 
+fail:
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/cfg_srv.c]: Let's rename it to "done" as in other places. We should free mbuf irrespective of the result.

File: nimble/host/mesh/src/friend.c [hunk 2]
```diff
@@ -453,6 +453,8 @@ static void enqueue_sub_cfm(struct bt_mesh_friend *frnd, u8_t xact)
 
 	frnd->last = buf;
 	frnd->send_last = 1;
+
+fail:
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/friend.c]: Again: "done"

File: nimble/host/mesh/src/friend.c [hunk 3]
```diff
@@ -760,6 +762,8 @@ static void enqueue_offer(struct bt_mesh_friend *frnd, s8_t rssi)
...
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/friend.c]: Again: "done"

File: nimble/host/mesh/src/transport.c [hunk 4]
```diff
@@ -590,8 +590,9 @@ static int sdu_recv(struct bt_mesh_net_rx *rx, u32_t seq, u8_t hdr,
 
 	if (IS_ENABLED(CONFIG_BT_MESH_FRIEND) && !rx->local_match) {
 		BT_DBG("Ignoring PDU for LPN 0x%04x of this Friend",
-		       rx->ctx.recv_dst);
-		return 0;
+			   rx->ctx.recv_dst);
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/transport.c]: The indentation was fine here.

**Comment by forgithub123** [comment]
Renamed 'fail' as 'done', and the indentation is my mistake.

**Comment by michal-narajowski** [comment]
@forgithub123 Please squash these two commits into one. You can use `git rebase -i master` and then follow help.

File: nimble/host/mesh/src/transport.c [hunk 5]
```diff
@@ -590,8 +590,9 @@ static int sdu_recv(struct bt_mesh_net_rx *rx, u32_t seq, u8_t hdr,
 
 	if (IS_ENABLED(CONFIG_BT_MESH_FRIEND) && !rx->local_match) {
 		BT_DBG("Ignoring PDU for LPN 0x%04x of this Friend",
-		       rx->ctx.recv_dst);
-		return 0;
+			rx->ctx.recv_dst);
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/transport.c]: This part still needs whitespace fix:

```suggestion
		       rx->ctx.recv_dst);
```

Comment (michal-narajowski) [reply nimble/host/mesh/src/transport.c]: In other words, mesh code was ported from Zephyr and the coding style should be the same. So you should use tabs and then align to the beginning of parenthesis with spaces.

**Pull request merged**